### PR TITLE
X953 multiple work numbers in block registration page

### PR DESCRIPTION
**Pull request by seenanair**
- Implementation of a multi select component
- Added work number field in Block Registration page

### CODE REVIEW

File: src/components/WorkNumberSelect.tsx [hunk 1]
```diff
@@ -100,78 +117,102 @@ export default function WorkNumberSelect({
...
+      ? selectedWork[selectedIndex >= 0 ? selectedIndex : selectedWork.length - 1]
+      : undefined
+    : selectedWork;
+
   const validateWorkNumber = () => {
-    if (!selectedWork) {
-      setError('SGP number is required');
+    if (multiple) {
+      if (!currentSelectedWork) {
+        setError('Atleast one work number must be selected');
```
Comment (khelwood) [inline src/components/WorkNumberSelect.tsx:163]: `'At least one work number must be selected.'

File: src/components/multiSelect/MultiSelect.tsx [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+  'block w-full py-2 px-3 border border-gray-300 bg-white rounded-md shadow-sm focus:outline-none focus:ring-sdb-100 focus:border-sdb-100 disabled:opacity-75 disabled:bg-gray-200 disabled:cursor-not-allowed';
+
+export const MultiSelect = React.forwardRef<HTMLSelectElement, SelectProps>(
+  ({ children, options, multiple = true, selectedValueProp, notifySelection, ...props }, ref) => {
+    const [selected, setSelected] = useState<Array<string>>([]);
+    const [selectionIndex, setSelectionIndex] = useState(-1);
+
+    React.useEffect(() => {
+      //Update selection, only if the given selection is different from what already exists
+      if (options.length < 0) return;
```
Comment (khelwood) [inline src/components/multiSelect/MultiSelect.tsx:43]: Wouldn't `options.length == 0` be more useful here? I think `options.length < 0` is impossible.

File: src/pages/BlockRegistration.tsx [hunk 3]
```diff
@@ -67,6 +68,7 @@ function getRegistrationFormTissue(): RegistrationFormTissue {
 function buildRegistrationSchema(registrationInfo: GetRegistrationInfoQuery) {
   const validation = new RegistrationValidation(registrationInfo);
   return Yup.object().shape({
+    workNumbers: Yup.array().min(1, 'Atleast one work number must be selected'),
```
Comment (khelwood) [inline src/pages/BlockRegistration.tsx:71]: `'At least one work number must be selected.'`

**Review by khelwood — CHANGES_REQUESTED** [review]
I've commented a couple of places for minor changes.

File: src/components/multiSelect/MultiSelect.tsx (hunk 2 above)
Comment (seenanair) [reply src/components/multiSelect/MultiSelect.tsx]: Yes, it's a mistake. Corrected

File: src/pages/BlockRegistration.tsx (hunk 3 above)
Comment (seenanair) [reply src/pages/BlockRegistration.tsx]: Done

**Review by khelwood — APPROVED** [review]

**Pull request merged**